File: mpi/pmpi/send.c
#include <stdio.h>
#include "mpi.h"

int main(int argc, char *argv[]) {
  int myid;
  int n_procs;

  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &myid);
  MPI_Comm_size(MPI_COMM_WORLD, &n_procs);

  if (n_procs < 2) {
    if (myid == 0) {
      fprintf(stderr, "n_procs must be bigger than 2\n");
    }
    MPI_Finalize();
    return -1;
  }

  int data = 0;
  MPI_Status status;

  if (myid == 0) {
    data = 100;
    MPI_Send(&data, 1, MPI_INT, 1, 1000, MPI_COMM_WORLD);
  } else if (myid == 1) {
    MPI_Recv(&data, 1, MPI_INT, 0, 1000, MPI_COMM_WORLD, &status);
  }

  printf("Rank[%d]: data=%d\n", myid, data);
  MPI_Finalize();
  return 0;
}

File: mpi/fortran/wrap_mpifunc00/send.f
      PROGRAM p2p
C Run with two processes
      include 'mpif.h'
      INTEGER err, rank, size
      real data(1)
      integer status(MPI_STATUS_SIZE)
      CALL MPI_INIT(err)
      CALL MPI_COMM_RANK(MPI_COMM_WORLD, rank, err)
      CALL MPI_COMM_SiZE(MPI_COMM_WORLD, size, err)

      data = 0.0
      if (rank.eq.0) then
         data=100.0
         call MPI_SEND(data, 1, MPI_REAL, 1, 55, MPI_COMM_WORLD, err)
      else if (rank.eq.1) then
         call MPI_RECV(data, 1, MPI_REAL, 0, 55,
     &        MPI_COMM_WORLD, status, err)
      end if
      print *, "Rank[", rank, "] data=",data

      call MPI_FINALIZE(err)
      end
